Packet++/src/WireGuardLayer.cpp

```diff
@@ -0,0 +1,324 @@
+			return new WireGuardHandshakeInitiationLayer(data, dataLen, prevLayer, packet);
+		case static_cast<uint8_t>(WireGuardMessageType::HandshakeResponse):
+			return new WireGuardHandshakeResponseLayer(data, dataLen, prevLayer, packet);
+		case static_cast<uint8_t>(WireGuardMessageType::CookieReply):
+			return new WireGuardCookieReplyLayer(data, dataLen, prevLayer, packet);
+		case static_cast<uint8_t>(WireGuardMessageType::TransportData):
+			return new WireGuardTransportDataLayer(data, dataLen, prevLayer, packet);
+		default:
+			return nullptr;
+		}
+	}
+
+	std::string WireGuardLayer::getMessageTypeAsString() const
+	{
+		// Assuming you have a method to retrieve the message type from the header
+		wg_common_header* wgHeader = reinterpret_cast<wg_common_header*>(this->getData());
+		switch (wgHeader->messageType)
+		{
+		case static_cast<uint8_t>(WireGuardMessageType::HandshakeInitiation):
+			return "Handshake Initiation";
+		case static_cast<uint8_t>(WireGuardMessageType::HandshakeResponse):
+			return "Handshake Response";
+		case static_cast<uint8_t>(WireGuardMessageType::CookieReply):
+			return "Cookie Reply";
+		case static_cast<uint8_t>(WireGuardMessageType::TransportData):
+			return "Transport Data";
+		default:
+			return "Unknown";
+		}
+	}
+
+	std::string WireGuardLayer::toString() const
+	{
+		return "WireGuardLayer, " + getMessageTypeAsString() + " message";
+	}
+
+	size_t WireGuardLayer::getHeaderLen() const
+	{
+		return m_DataLen;
+	}
+
+	bool WireGuardLayer::isDataValid(const uint8_t* data, size_t dataLen)
+	{
+		if (dataLen < sizeof(WireGuardLayer::wg_common_header))
+			return false;
+
+		uint8_t messageType = data[0];
+		return messageType >= static_cast<uint8_t>(WireGuardLayer::WireGuardMessageType::HandshakeInitiation) &&
+		       messageType <= static_cast<uint8_t>(WireGuardLayer::WireGuardMessageType::TransportData);
+	}
+
+	// ~~~~~~~~~~~~~~~~~~~~
+	// WireGuardHandshakeInitiationLayer
+	// ~~~~~~~~~~~~~~~~~~~~
+
+	WireGuardHandshakeInitiationLayer::WireGuardHandshakeInitiationLayer(uint32_t senderIndex,
+	                                                                     const uint8_t initiatorEphemeral[32],
+	                                                                     const uint8_t encryptedInitiatorStatic[48],
+	                                                                     const uint8_t encryptedTimestamp[28],
+	                                                                     const uint8_t mac1[16], const uint8_t mac2[16])
+	{
+		const size_t messageLen = sizeof(wg_handshake_initiation);
+		m_DataLen = messageLen;
+		m_Data = new uint8_t[messageLen];
+		memset(m_Data, 0, messageLen);
+
+		wg_handshake_initiation* msgHdr = reinterpret_cast<wg_handshake_initiation*>(m_Data);
+
+		msgHdr->messageType = static_cast<uint8_t>(WireGuardMessageType::HandshakeInitiation);
+
+		std::memset(msgHdr->reserved, 0, sizeof(msgHdr->reserved));
```

Comment:
This is redundant because we already use `memset` in line 95.

Same in the other c'tors